processor/transformprocessor/internal/metrics/func_extract_sum_metric_test.go

```diff
@@ -0,0 +1,336 @@
+	input := metricInput.Summary().DataPoints().AppendEmpty()
+	input.SetCount(100)
+	input.SetSum(12.34)
+
+	qVal1 := input.QuantileValues().AppendEmpty()
+	qVal1.SetValue(1)
+	qVal1.SetQuantile(.99)
+
+	qVal2 := input.QuantileValues().AppendEmpty()
+	qVal2.SetValue(2)
+	qVal2.SetQuantile(.95)
+
+	qVal3 := input.QuantileValues().AppendEmpty()
+	qVal3.SetValue(3)
+	qVal3.SetQuantile(.50)
+
+	attrs := getTestAttributes()
+	attrs.CopyTo(input.Attributes())
+	return metricInput
+}
+
+func getTestGaugeMetric() pmetric.Metric {
+	metricInput := pmetric.NewMetric()
+	metricInput.SetEmptyGauge()
+	metricInput.SetName("gauge_metric")
+	input := metricInput.Gauge().DataPoints().AppendEmpty()
+	input.SetIntValue(12)
+
+	attrs := getTestAttributes()
+	attrs.CopyTo(input.Attributes())
+	return metricInput
+}
+
+func getTestAttributes() pcommon.Map {
+	attrs := pcommon.NewMap()
+	attrs.PutStr("test", "hello world")
+	attrs.PutInt("test2", 3)
+	attrs.PutBool("test3", true)
+	return attrs
+}
+
+type histogramTestCase struct {
+	name         string
+	input        pmetric.Metric
+	temporality  pmetric.AggregationTemporality
+	monotonicity bool
+	want         func(pmetric.MetricSlice)
+}
+
+func Test_extractSumMetric(t *testing.T) {
+	tests := []histogramTestCase{
+		{
+			name:         "histogram",
+			input:        getTestHistogramMetric(),
+			temporality:  pmetric.AggregationTemporalityDelta,
+			monotonicity: false,
+			want: func(metrics pmetric.MetricSlice) {
+				histogramMetric := getTestHistogramMetric()
+				histogramMetric.CopyTo(metrics.AppendEmpty())
+				sumMetric := metrics.AppendEmpty()
+				sumMetric.SetEmptySum()
+				sumMetric.Sum().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+				sumMetric.Sum().SetIsMonotonic(false)
+
+				sumMetric.SetName(histogramMetric.Name() + "_sum")
+				dp := sumMetric.Sum().DataPoints().AppendEmpty()
+				dp.SetDoubleValue(histogramMetric.Histogram().DataPoints().At(0).Sum())
+
+				attrs := getTestAttributes()
+				attrs.CopyTo(dp.Attributes())
+			},
+		},
+		{
+			name:         "histogram (monotonic)",
+			input:        getTestHistogramMetric(),
+			temporality:  pmetric.AggregationTemporalityDelta,
+			monotonicity: true,
+			want: func(metrics pmetric.MetricSlice) {
+				histogramMetric := getTestHistogramMetric()
+				histogramMetric.CopyTo(metrics.AppendEmpty())
+				sumMetric := metrics.AppendEmpty()
+				sumMetric.SetEmptySum()
+				sumMetric.Sum().SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+				sumMetric.Sum().SetIsMonotonic(true)
+
+				sumMetric.SetName(histogramMetric.Name() + "_sum")
+				dp := sumMetric.Sum().DataPoints().AppendEmpty()
+				dp.SetDoubleValue(histogramMetric.Histogram().DataPoints().At(0).Sum())
+
+				attrs := getTestAttributes()
+				attrs.CopyTo(dp.Attributes())
+			},
+		},
+		{
+			name:         "histogram (cumulative)",
+			input:        getTestHistogramMetric(),
+			temporality:  pmetric.AggregationTemporalityCumulative,
```

Comment:
Can you add a test for cumulative and monotonic?  And if that isn't a valid combination, let's validate that in the function.

Reply:
Sure, can do. The combination is valid per the spec, these attributes are independent of each other.

Reply:
Done: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24368/commits/ac9e282ebc2079914e71b9524d13bb20a93d9260